src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -691,9 +691,17 @@ public override void VisitBinaryOperationSyntax(BinaryOperationSyntax syntax)
                     return operatorInfo.ReturnType;
                 }
 
+                string? additionalInfo = null;
+                if (operandType1.TypeKind is TypeKind.StringLiteral &&
+                    operandType2.TypeKind is TypeKind.StringLiteral &&
+                    syntax.Operator is BinaryOperator.Add)
```

Comment:
Is it deliberate that this is limited to `StringLiteralType` (not regular string)?

Reply:
`StringLiteralType` is different than the enum `TypeKind.StringLiteral `. From what I saw, variables (literal/object access), params and string literals all constitute as `StringLiteral` whereas interpolated strings end up being `TypeKind.Primitive`.
![image](https://user-images.githubusercontent.com/3432839/140980080-ef26b114-22a7-46e1-a13a-9fcb03157d0b.png)


In that case, its sufficient to assume the user can understand their mistake since they have a interpolated string as one of the two operand's arguments, so behavior is unchanged.
![image](https://user-images.githubusercontent.com/3432839/140979982-51a8586d-9b5a-4d92-9aed-7312e8398a83.png)

Reply:
`StringLiteral` will *only* represent literal strings (`'abc'`). I think you're forgetting that regular strings can come from places other than interpolation - e.g. parameters, accessing resource/module properties, function return values etc:

```bicep
param myString string

var joined = myString + resourceGroup().location
```

Reply:
It's not obvious why the + operator can't operate on strings even when the two types are `string` and `string` instead of literal string types, so I think we need to check if either or both of the operands are assignable to `string`.

Reply:
Yeah, agreed

Reply:
Thanks for the catch @anthony-c-martin, following the advice I have added checks for `string` types as well.
`(operandType1.TypeKind is TypeKind.Primitive && string.Equals(operandType1.Name, LanguageConstants.TypeNameString))`

Reply:
You could actually do this instead to reuse the type check logic, but this is fine too: `TypeValidator.AreTypesAssignable(operandType1, LanguageConstants.String) && ...

Reply:
Thanks! That makes it cleaner, done 👍🏾